python/paddle/fluid/tests/unittests/ir/test_ir_graph_to_program_pass.py

```diff
@@ -0,0 +1,205 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import unittest
+import six
+import paddle
+from paddle import fluid
+
+paddle.enable_static()
+
+
+def program_to_IRGraph(program):
+    graph = fluid.core.Graph(program.desc)
+    ir_graph = fluid.framework.IrGraph(graph, for_test=False)
+    return ir_graph
+
+
+def IRGraph_to_program(ir_graph):
+    return ir_graph.to_program()
+
+
+class GraphToProgramPassTest(unittest.TestCase):
+    def check_vars_equal(self, o_block, c_block):
+        o_params = sorted(o_block.all_parameters(), key=lambda p: p.name)
+        c_params = sorted(c_block.all_parameters(), key=lambda p: p.name)
+        self.assertEqual(len(o_params), len(c_params))
+        for p_idx in range(len(o_params)):
+            self.assertEqual(o_params[p_idx].name, c_params[p_idx].name)
+
+        o_vars = sorted(o_block.vars.values(), key=lambda v: v.name)
+        c_vars = sorted(c_block.vars.values(), key=lambda v: v.name)
+        self.assertEqual(len(o_vars), len(c_vars))
+        for v_idx in range(len(o_vars)):
+            self.assertEqual(o_vars[v_idx].name, c_vars[v_idx].name)
+
+    def check_op_output_equal(self, o_op, c_op):
+        self.assertEqual(len(o_op.output_names), len(c_op.output_names))
+        for out_idx in range(len(o_op.output_names)):
+            o_out = o_op.output_names[out_idx]
+            c_out = c_op.output_names[out_idx]
+            self.assertEqual(o_out, c_out)
+            self.assertEqual(o_op.output(o_out), c_op.output(c_out))
+
+    def check_op_input_equal(self, o_op, c_op):
+        self.assertEqual(len(o_op.input_names), len(c_op.input_names))
+        for in_idx in range(len(o_op.input_names)):
+            o_in = o_op.input_names[in_idx]
+            c_in = c_op.input_names[in_idx]
+            self.assertEqual(o_in, c_in)
+            self.assertEqual(o_op.input(o_in), c_op.input(c_in))
+
+    def check_op_attrs_equal(self, o_op, c_op):
+        o_attrs = sorted(o_op.attr_names)
+        c_attrs = sorted(c_op.attr_names)
+        self.assertEqual(len(o_attrs), len(c_attrs))
+        for attr_idx in range(len(o_attrs)):
+            o_attr = o_attrs[attr_idx]
+            c_attr = c_attrs[attr_idx]
+            self.assertEqual(o_attr, c_attr)
+            self.assertEqual(
+                o_op.desc.attr_type(o_attr), c_op.desc.attr_type(c_attr))
+
+
+class SingleGraphToProgramPass(GraphToProgramPassTest):
+    def setUp(self):
+        self.origin_program = self.build_program()
+        ir_graph = program_to_IRGraph(self.origin_program)
+        self.converted_program = IRGraph_to_program(ir_graph)
+
+    @staticmethod
+    def build_program():
+        program = fluid.default_main_program()
```

Comment:
Please write 2.0 API instead of fluid API for the test cases in the future.

Reply:
Done